plugin/trino-geospatial/pom.xml

```diff
@@ -33,23 +33,26 @@
         </dependency>
 
         <dependency>
-            <groupId>com.fasterxml.jackson.core</groupId>
-            <artifactId>jackson-databind</artifactId>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
         </dependency>
 
         <dependency>
-            <groupId>com.google.guava</groupId>
-            <artifactId>guava</artifactId>
+            <groupId>org.locationtech.jts</groupId>
+            <artifactId>jts-core</artifactId>
         </dependency>
 
+        <!-- used by tests but also needed transitively -->
         <dependency>
-            <groupId>it.unimi.dsi</groupId>
-            <artifactId>fastutil</artifactId>
+            <groupId>com.fasterxml.jackson.core</groupId>
+            <artifactId>jackson-databind</artifactId>
+            <scope>runtime</scope>
         </dependency>
 
         <dependency>
-            <groupId>org.locationtech.jts</groupId>
-            <artifactId>jts-core</artifactId>
+            <groupId>it.unimi.dsi</groupId>
+            <artifactId>fastutil</artifactId>
```

Comment:
comes from trino-array. databind comes from io.airlift.json.

But all of these are `compile` scope in the dependencies which pull these transitively. So why do we need to specify runtime here?

Also trino-array is used in compiled code, not tests so fastutil should also be compile scoped.

Reply:
It's not used directly in the main code but in the test code. So when you add this to the compile scope, plugin will report that you are using test-only dependency in the broarder scope (compile). That's why it's moved to runtime.